windows-agent/internal/proservices/landscape/executor.go

```diff
@@ -305,6 +322,52 @@ func download(ctx context.Context, f io.Writer, url, checksum string) (err error
 	return nil
 }
 
+// wantRootfsChecksum fetches the checksum from the SHA256SUMS file found alongside the rootfs URL matching the rootfs file name.
+//
+// The SHA256SUMS file is expected to contain multiple lines of the format:
+//
+// SHA256 *filename
+//
+// For example:
+//
+// 03c7f7c75fb450c7dd576a0da20986e62e0d72bd2ccee4c01296bab9f415c7ab *jammy-server-cloudimg-amd64-azure.vhd.tar.gz
+// 0dc4d78f08e871ce6325e027e1b8421fd1cde1e76158644e35343a36d8f67bf4 *jammy-server-cloudimg-amd64-root.tar.xz
+// 103ee8b5693bdb7c23a378453c624d8605445eb07e2e550d3fad831da865f5ea *jammy-server-cloudimg-riscv64.release.20240514.20240601.image_changelog.json
+// 1eaa1df5794122e3419c963d88f043121c164936b9b828adac650c9f5e22c3e6 *jammy-server-cloudimg-amd64.img
+// 1fcd2edf4fda78e0a6f3bc0c3684286c29371e4dd7863a59b39d2cfcff79b5e1 *jammy-server-cloudimg-amd64-root.manifest
+// 1fcd2edf4fda78e0a6f3bc0c3684286c29371e4dd7863a59b39d2cfcff79b5e1 *jammy-server-cloudimg-amd64.squashfs.manifest
+// 2646292d657f4c9ef5dfce804a5a1e66d8c1324c74147b8bc9b1bf154d7feaf8 *jammy-server-cloudimg-arm64-root.tar.xz
+//
+// ...
+func wantRootfsChecksum(u *url.URL) (string, error) {
+	imageName := filepath.Base(u.Path)
+	shasRelativeLocation, err := url.Parse("../SHA256SUMS")
+	if err != nil {
+		return "", fmt.Errorf("could not assemble SHA256SUMS location: %v", err)
+	}
+	checksumsURL := u.ResolveReference(shasRelativeLocation)
+
+	resp, err := http.Get(checksumsURL.String())
+	if err != nil {
+		return "", fmt.Errorf("could not download checksums file <%s>: %v", checksumsURL, err)
```

Comment:
rather than <%s>, you can use %q so that it’s quoted, which is more standard I think than having it with `<>`?